pkg/apis/core/types.go

```diff
@@ -2434,6 +2434,10 @@ const (
 	// disruption (such as preemption, eviction API or garbage-collection).
 	// The constant is to be renamed once the name is accepted within the KEP-3329.
 	AlphaNoCompatGuaranteeDisruptionTarget PodConditionType = "DisruptionTarget"
+	// ResourceExhausted indicates the pod is about to be deleted due to either
+	// exceeding its ephemeral storage limits or running an OOM killed container.
+	// The constant is to be renamed once the name is accepted within the KEP-3329.
+	AlphaNoCompatGuaranteeResourceExhausted = "ResourceExhausted"
```

Comment:
since apiserver doesn't need this constant, it doesn't need to be defined in this file, only the versioned one.

Reply:
We want it eventually here and it might be also needed by the disruption controller to clean up stale ones. I think it is possible that a pod exceeds (temporarily) ephemeral storage limit, adding the condition succeeds, but deleting fails. Then, I think kubelet wouldn't guarantee retrying the delete in the future IIUC.

Reply:
the disruption controller is not in apiserver, so it should use the versioned packages

Reply:
I think the comment needs more clarity around whether the condition is added when a container fails, even if it's going to be retried.

Reply:
true, actually I think we may not need the change in the disruption controller too, please take a look at another comment